package-lock.json

```diff
@@ -272,6 +272,21 @@
       "resolved": "https://registry.npmjs.org/acorn/-/acorn-5.2.1.tgz",
       "integrity": "sha512-jG0u7c4Ly+3QkkW18V+NRDN+4bWHdln30NL1ZL2AvFZZmQe/BfopYCtghCKKVBUSetZ4QKcyA0pY6/4Gw8Pv8w=="
```

Comment:
Should we add this file to gitignore?

Reply:
Maybe. I'm not entirely what the best practice is here, since these files are relatively new in the npm world, afaik.

On the one hand, this guarantees we're all using the same exact versions of all dependencies when running our dev builds. On the other hand, this file is going through a lot of churn in the initial phases of the project as we add and remove dependencies.

@samdutton @rowan-m What do you think?

Reply:
I think it is autogenerated though? This means from the same package.json we should get the same lock? Or maybe I'm wrong?

Reply:
They're auto-generated, but they're fixed to a particular point in time. The same `package.json` will generate a different `package-lock.json` depending on the current version of dependencies and sub-dependencies on NPM. So I'd be pulling in a different set of packages if I did an `npm i` today than if I did so tomorrow. With a `package-lock.json` the versions are exactly the same, unless `package-lock.json` itself changes.

Reply:
Seems like worth checking in then.

Reply:
https://docs.npmjs.com/files/package-lock.json
> "This file is intended to be committed into source repositories"